### PR TITLE
Ros2 unify filenames

### PR DESCRIPTION
**Pull request by pawelirh**
### Description

This PR renames and restructures the files

### Modifications

- Rename some of the files to improve readability
- Create subdirectories to store subdomain-related code
- Update include guards according best guidelines

## Summary by CodeRabbit

- **New Features**
	- Updated configuration file paths for improved clarity and organization.
	- Renamed ROS nodes to enhance understanding of their functionalities.
	- Introduced a utility class for managing testing lifecycle and URDF generation.
- **Bug Fixes**
	- Adjusted descriptions in the README to better reflect node functionalities.
- **Documentation**
	- Enhanced documentation to reflect file renaming and new structures, including updated links for easier navigation.
- **Chores**
	- Reorganized test configurations and files for better structure.
	- Updated include guards for consistency across header files.

### CODE REVIEW

File: ROS_API.md [hunk 1]
```diff
@@ -42,23 +42,23 @@ Below is information about the physical robot API. For the simulation, topics an
...
-| 🤖   | `battery_driver`          | Publishes battery state read from ADC unit for Panther version 1.2 and above, or based on Roboteq motor controllers' data for earlier versions of the robot. <br/> [*panther_batter/battery_node*](./panther_batter)                                                                                                                                                                                                                                                                                                                  |
+| 🤖   | `battery_driver`          | Publishes battery state read from ADC unit for Panther version 1.2 and above, or based on Roboteq motor controllers' data for earlier versions of the robot. <br/> [*panther_batter/battery_driver_node*](./panther_battery/src/battery_driver_node.cpp)                                                                                                                                                                                                                                                                                     |
```
Comment (KmakD) [inline ROS_API.md]: Shouldn't we rather point to header files?

File: panther_battery/test/test_battery_driver_node_adc_single.cpp
Comment (KmakD) [inline panther_battery/test/test_battery_driver_node_adc_single.cpp:1]: It also tests RoboteqInitOnADCFail, so I'm not sure if it's `_adc_single`

File: panther_diagnostics/config/system_monitor.yaml [hunk 2]
```diff
@@ -7,12 +7,12 @@ system_status:
     type: string
     default_value: Built-in Computer
     description: Name of hardware used in diagnostics.
-  publish_rate:
+  publish_frequency:
     type: double
-    default_value: 0.25
-    description: System status publish rate in seconds.
+    default_value: 5.0
```
Comment (KmakD) [inline panther_diagnostics/config/system_monitor.yaml:12]: Why are we changing frequency?

File: panther_diagnostics/src/system_monitor_node.cpp [hunk 3]
```diff
@@ -12,57 +12,60 @@
...
 
   param_listener_ =
-    std::make_shared<system_status::ParamListener>(this->get_node_parameters_interface());
+    std::make_shared<system_monitor::ParamListener>(this->get_node_parameters_interface());
   params_ = param_listener_->get_params();
 
   system_status_publisher_ = this->create_publisher<panther_msgs::msg::SystemStatus>(
     "system_status", 10);
+
+  auto timer_interval_ms = static_cast<long long>(1000.0 / params_.publish_frequency);
```
Comment (KmakD) [inline panther_diagnostics/src/system_monitor_node.cpp]:
```suggestion
  const auto timer_interval_ms = static_cast<long long>(1000.0 / params_.publish_frequency);
```

File: panther_hardware_interfaces/include/panther_hardware_interfaces/panther_system/panther_system_e_stop.hpp [hunk 4]
```diff
@@ -12,16 +12,16 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-#ifndef PANTHER_HARDWARE_INTERFACES_PANTHER_SYSTEM_E_STOP_MANAGER_HPP_
-#define PANTHER_HARDWARE_INTERFACES_PANTHER_SYSTEM_E_STOP_MANAGER_HPP_
+#ifndef PANTHER_HARDWARE_INTERFACES_PANTHER_SYSTEM_PANTHER_SYSTEM_E_STOP_MANAGER_HPP_
+#define PANTHER_HARDWARE_INTERFACES_PANTHER_SYSTEM_PANTHER_SYSTEM_E_STOP_MANAGER_HPP_
```
Comment (KmakD) [inline panther_hardware_interfaces/include/panther_hardware_interfaces/panther_system/panther_system_e_stop.hpp]:
```suggestion
#ifndef PANTHER_HARDWARE_INTERFACES_PANTHER_SYSTEM_PANTHER_SYSTEM_E_STOP_HPP_
#define PANTHER_HARDWARE_INTERFACES_PANTHER_SYSTEM_PANTHER_SYSTEM_E_STOP_HPP_
```

File: panther_hardware_interfaces/include/panther_hardware_interfaces/panther_system/panther_system_e_stop.hpp [hunk 5]
```diff
@@ -188,4 +188,4 @@ class EStopPTH10X : public EStopInterface
 
 }  // namespace panther_hardware_interfaces
 
-#endif  // PANTHER_HARDWARE_INTERFACES_PANTHER_SYSTEM_E_STOP_MANAGER_HPP_
+#endif  // PANTHER_HARDWARE_INTERFACES_PANTHER_SYSTEM_PANTHER_SYSTEM_E_STOP_MANAGER_HPP_
```
Comment (KmakD) [inline panther_hardware_interfaces/include/panther_hardware_interfaces/panther_system/panther_system_e_stop.hpp]:
```suggestion
#endif  // PANTHER_HARDWARE_INTERFACES_PANTHER_SYSTEM_PANTHER_SYSTEM_E_STOP_HPP_
```

File: panther_manager/CMakeLists.txt [hunk 6]
```diff
@@ -96,52 +96,51 @@ install(DIRECTORY behavior_trees config launch
 
 if(BUILD_TESTING)
   find_package(ament_cmake_gtest REQUIRED)
+  find_package(PkgConfig REQUIRED)
+  pkg_check_modules(LIBSSH REQUIRED libssh)
+  find_package(yaml-cpp REQUIRED)
```
Comment (KmakD) [inline panther_manager/CMakeLists.txt]: This dependencies are already included in PACKAGE_DEPENDENCIES

**Review by KmakD — CHANGES_REQUESTED** [review]

File: panther_diagnostics/config/system_monitor.yaml (hunk 2 above)
Comment (pawelirh) [reply panther_diagnostics/config/system_monitor.yaml:12]: For time intervals 250Hz is an intuitive default value, for frequency 4Hz is not, and that's the only reason. This is why I increased it a bit to be intuitive but still clocking roughly the same.

**Comment by rafal-gorecki** [comment]
Panther manager has a plugins folder which contains plugins, and panther hardware interfaces contains plugins inside src. This seems inconsistent.

File: panther_battery/test/test_battery_driver_node_adc_single.cpp
Comment (pawelirh) [reply panther_battery/test/test_battery_driver_node_adc_single.cpp:1]: Correct, and we do this test on a single battery test fixture object.

**Comment by pawelirh** [comment]
> Panther manager has a plugins folder which contains plugins, and panther hardware interfaces contains plugins inside src. This seems inconsistent.

As for the `panther_manager` package, the `plugin` directory aims to distinguish bt nodes (technically called plugins) among other files in this package. Compared to `panther_hardware_interfaces`, this package stores only plugins. Adding here subfolder `plugins` IMO would nest the structure deeper without benefits.

What I can propose if we want to make the structure more cohesive among those two packages:
- Move the `plugins` folder with `.cpp` files under the `src` folder. (.hpp and test files stay untouched)
- [ADDITIONAL NOTE] In `panther_hardware_interfaces/test` we got utility code distributed among `include` and `src` folders. We can implement those utilities in header files to remove `.cpp` files and (same as in `panther_manager`) create a `test/utils` folder to store them.

 @rafal-gorecki @KmakD Join the discussion ;)

**Review by KmakD — APPROVED** [review]

**Pull request merged**